src/Microsoft.Android.Build.BaseTasks/AsyncTask.cs

```diff
@@ -0,0 +1,431 @@
+		}
+
+		public void Cancel() => taskCancelled.Set();
+
+		protected void Complete(System.Threading.Tasks.Task task)
+		{
+			if (task.Exception != null)
+			{
+				var ex = task.Exception.GetBaseException();
+				LogError(ex.Message + Environment.NewLine + ex.StackTrace);
+			}
+			Complete();
+		}
+
+		public void Complete() => completed.Set();
+
+		public void LogDebugTaskItems(string message, string[] items)
+		{
+			LogDebugMessage(message);
+
+			if (items == null)
+				return;
+
+			foreach (var item in items)
+				LogDebugMessage("	{0}", item);
+		}
+
+		public void LogDebugTaskItems(string message, ITaskItem[] items)
+		{
+			LogDebugMessage(message);
+
+			if (items == null)
+				return;
+
+			foreach (var item in items)
+				LogDebugMessage("	{0}", item.ItemSpec);
+		}
+
+		public void LogTelemetry(string eventName, IDictionary<string, string> properties)
+		{
+			if (uiThreadId == Thread.CurrentThread.ManagedThreadId)
+			{
+#pragma warning disable 618
+				Log.LogTelemetry(eventName, properties);
+				return;
+#pragma warning restore 618
+			}
+
+			var data = new TelemetryEventArgs() {
+					EventName = eventName,
+					Properties = properties
+			};
+			EnqueueMessage(telemetryMessageQueue, data, telemetryDataAvailable);
+		}
+
+		public void LogMessage(string message) => LogMessage(message, importance: MessageImportance.Normal);
+
+		public void LogMessage(string message, params object[] messageArgs) => LogMessage(string.Format(message, messageArgs));
+
+		public void LogDebugMessage(string message) => LogMessage(message, importance: MessageImportance.Low);
+
+		public void LogDebugMessage(string message, params object[] messageArgs) => LogMessage(string.Format(message, messageArgs), importance: MessageImportance.Low);
+
+		public void LogMessage(string message, MessageImportance importance = MessageImportance.Normal)
+		{
+			if (uiThreadId == Thread.CurrentThread.ManagedThreadId)
+			{
+#pragma warning disable 618
+				Log.LogMessage(importance, message);
+				return;
+#pragma warning restore 618
+			}
+
+			var data = new BuildMessageEventArgs(
+					message: message,
+					helpKeyword: null,
+					senderName: null,
+					importance: importance
+			);
+			EnqueueMessage(logMessageQueue, data, logDataAvailable);
+		}
+
+		public void LogError(string message) => LogCodedError(code: null, message: message, file: null, lineNumber: 0);
+
+		public void LogError(string message, params object[] messageArgs) => LogCodedError(code: null, message: string.Format(message, messageArgs));
+
+		public void LogErrorFromException(Exception exception)
+		{
+			if (uiThreadId == Thread.CurrentThread.ManagedThreadId)
+			{
+#pragma warning disable 618
+				Log.LogErrorFromException(exception);
+				return;
+#pragma warning restore 618
+			}
+
+			StackFrame exceptionFrame = null;
+			try
+			{
+				exceptionFrame = new StackTrace(exception, true)?.GetFrames()?.FirstOrDefault();
+			}
+			catch { }
```

Comment:
Looks like this will only throw if exception is null, so I added a null check and removed the try/catch https://learn.microsoft.com/en-us/dotnet/api/system.diagnostics.stacktrace.-ctor?view=net-8.0#system-diagnostics-stacktrace-ctor(system-exception-system-boolean)